saleor/product/models.py

```diff
@@ -478,6 +478,18 @@ def get_assigned_attributes(self, product_type_pk: int):
             | Q(attributevariant__product_type_id=product_type_pk)
         )
 
+    @staticmethod
+    def user_has_access_to_all(user):
+        return user.is_active and user.has_perm("product.manage_products")
+
+    def get_public_attributes(self):
+        return self.filter(visible_in_storefront=True)
+
+    def get_visible_to_user(self, user):
```

Comment:
Are you going to use `get_public_attributes`, `user_has_access_to_all`  somewhere else? If not I think that you can leave only `get_visible_to_user` method.
And another important question is: Are you sure that we want to show different attributes for admin users and for the rest of customers? It could lead to a strange situation like - admin user see newly added attributes, he checks storefront and everything looks good, but normal customers don't see newly added attributes.

Reply:
Yes, it will be used in another task

Reply:
> Are you sure that we want to show different attributes for admin users and for the rest of customers? It could lead to a strange situation like - admin user see newly added attributes, he checks storefront and everything looks good, but normal customers don't see newly added attributes.

In another task we will add filtering. The storefront will take care of displaying the correct information depending on the context.